tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -245,6 +245,7 @@ struct PythonTableArrowArrayStream {
 
 	static int MyStreamGetSchema(struct ArrowArrayStream *stream, struct ArrowSchema *out) {
 		D_ASSERT(stream->private_data);
+		py::gil_scoped_acquire acquire;
```

Comment:
I realise that this change stops it crashing, but we should double-check why that is the case.

Reply:
by double-check I mean find out